File: Lecture 5/practice/SelectionSort.py
def selectionSort(array):
    r = array
    for i in range(0, len(r)):
        for j in range(i + 1, len(r)):
            min = r[i]
            if min > r[j]:
                temp = r[i]
                r[i] = r[j]
                r[j] = temp
    return r

def printArray(r):
    print(*r)


array = ["H", "O", "M", "E", "W", "R", "K"]
print("Alphabetical order of 'H', 'O', 'M', 'E', 'W', 'R','K' is:")
r = selectionSort(array)
printArray(r)

#Result is "E H K M O R W"

File: Lecture 2/SelectionSort.java
import java.util.Scanner;

public class SelectionSort {
    
    public static Scanner sc = new Scanner(System.in);

    private static int[] inputArray(int a[]) {
		for(int i = 0; i < a.length; i++) {
			a[i] = sc.nextInt();
		}
		return a;
	}

    private static void swap(int a[], int i, int j) {
        int temp = a[i];
        a[i] = a[j];
        a[j] = temp;
    }

    public static void selectionSort(int[] a) {
        for (int i = 0; i < a.length; i++) {
            for (int j = i + 1; j < a.length; j++) {
                int min = a[i];
                if (min > a[j]) {
                    swap(a, i, j);
                }
            }
        }
    }

    public static void printArray(int[] a){
        for(int i = 0; i < a.length; i++){
            System.out.println(a[i] + " ");
        }
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();
        System.out.print("Number of elements: ");
            int n = sc.nextInt();
        int[] array = new int[n];
        inputArray(array);
        selectionSort(array);
        long endTime = System.currentTimeMillis();
        long totalTime = endTime - startTime;
        System.out.println();
        System.out.println("Running time: " + totalTime + "ms");
    }
}
